src/execution/index/art/art_index.cpp

```diff
@@ -0,0 +1,189 @@
+#include "duckdb/execution/index/art/art.hpp"
+#include "duckdb/execution/index/index_type.hpp"
+#include "duckdb/catalog/catalog_entry/duck_table_entry.hpp"
+#include "duckdb/catalog/catalog_entry/duck_index_entry.hpp"
+#include "duckdb/catalog/catalog_entry/table_catalog_entry.hpp"
+#include "duckdb/execution/index/art/art_key.hpp"
+#include "duckdb/execution/index/bound_index.hpp"
+#include "duckdb/main/client_context.hpp"
+#include "duckdb/main/database_manager.hpp"
+#include "duckdb/storage/storage_manager.hpp"
+#include "duckdb/storage/table/append_state.hpp"
+#include "duckdb/common/exception/transaction_exception.hpp"
+#include "duckdb/execution/index/art/art_operator.hpp"
+
+namespace duckdb {
+namespace {
+
+//----------------------------------------------------------------------------------------------------------------------
+// Bind
+//----------------------------------------------------------------------------------------------------------------------
+class ARTBuildBindData : public IndexBuildBindData {
+public:
+	bool sorted = false;
+};
+
+unique_ptr<IndexBuildBindData> ARTBuildBind(IndexBuildBindInput &input) {
+	auto bind_data = make_uniq<ARTBuildBindData>();
+
+	// TODO: Verify that the the ART is applicable for the given columns and types.
```

Comment:
Is this a `TODO` for this PR? Is this the type check we currently have in the `ART` constructor? Would be great if we can move that here. If I understand correctly, this would be another callback defined by the index type?